app/src/main/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsViewModel.kt

```diff
@@ -50,8 +51,47 @@ class TransactionDetailsViewModel
         @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
         set
 
+    fun submitRejection() {
+        val executionInfo = txDetails?.detailedExecutionInfo as? DetailedExecutionInfo.MultisigExecutionDetails
+            ?: throw RuntimeException("MissingCorrectExecutionDetailsException")
```

Comment:
I think here you could have a `class MissingCorrectExecutionDetailsExpect : Exception` instead, and leverage type safety at the UI level to display specific error messages

Reply:
AFAICT, we're not catching this anywhere. But the Button is hidden on non-eligible transactions. I don't mind making this a specific type.